telemetry/specs/INT.mdk

```diff
@@ -652,7 +652,40 @@ both INT header stacks carry information for respective layers and need not be
 considered interfering with each other.
 
 A field in Ethernet, IP, or TCP/UDP should indicate if the
-INT header exists after the TCP/UDP header. We propose two options.
+INT header exists after the TCP/UDP header. We propose three options.
+
+* UDP destination port field: a new UDP port number (INT_TBD) will be assigned
+by IANA to indicate the existence of INT after UDP. This option supports two
+use cases:
+  - The original packet already has UDP header either as user application
+    protocol or as part of another UDP-based encapsulation such as VXLAN,
+    GENEVE, RoCEv2. INT is inserted after the UDP header with the UDP
+    destination port number changed to INT_TBD.
+    The original destination port number
+    is carried in the shim header for the INT sink to restore back when it
+    removes the INT stack from the packet.
+  - A new UDP header for INT is inserted between IP and the existing L4 header.
+    The protocol field of IP header is set to 17 for UDP and the original
+    IP protocol value is carried in the INT shim header.
+    In the new UDP header, INT_TBD is used as the destination port number.
+    The source port number of the new UDP header must be either 1) copied from the
+    original L4 header, 2) computed as a hash of the existing L4 header
+    fields (such as source/destination ports in case of TCP/UDP), or 3)
+    computed as a hash of the original outer 5 tuple. The second option
+    is recommended to provide ECMP/LAG load-balancing logic with the same degree of
+    entropy of the original data packet.
```

Comment:
We should also add some language about the UDP checksum:

It is recommended that the checksum in the new UDP header be set to zero. For IPv6 packets, this falls under the case of tunnel protocols, which are allowed to use zero UDP checksums as specified in RFC 6935. The existing L4 header will typically include a checksum computed using the encapsulating IPv6 header fields, thus offering some protection against IPv6 header corruption.

Reply:
@mickeyspiegel I addressed the first comment. As for the UDP checksum, do we need additional language on top of the existing texts under "Checksum Update" section? Line 561-.

Reply:
That section is about checksum update, when the received packet included a checksum. In that case, if the received UDP checksum is zero, then there is no checksum update. If the received UDP checksum is non-zero, then the checksum must be updated in one of the ways specified in that section.

This case differs a little, since in this case we are adding the UDP header. In this case, the options are to either insert a UDP checksum with value 0, or compute a valid UDP checksum.

I don't care whether we cover this here, or add some text to section 5.5, but it should be discussed somewhere.

Reply:
Added some text here.